Including all required headings on README.md

heat-jr opened the pull request:
This still needs to be formatted. Will follow up later 11/2.

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

heat-jr committed:
Including all required headings on README.md

This still needs to be formatted. Will follow up later 11/2.

heat-jr committed:
Update README.md to include all template headings

Copy updates and added first screenshot under deployment guidance.

heat-jr committed:
Update content and formatting for README.md 

Revised procedures inline with AWS Style to "Train the RL model".

heat-jr committed:
Update README.md

Fixed line spacing.

davidfsmith commented on README.md:
```diff
@@ -1,10 +1,326 @@
+If you wish to create your own reward function there is a pattern to the function that you must use.
+
+
+12345def reward_function(params) :
+
+    reward = ...
+
+    return float(reward)
```
Check the syntax of this code element, during review it wasn't showing correctly as a code sample

heat-jr replied:
Done.

davidfsmith commented on README.md:
```diff
@@ -1,10 +1,326 @@
+Additionally, modify the hyperparameters in the notebook (Look two cells below the label Train the RL model using the Python SDK Script mode) to include the SAC hyperparameters instead:
+
+
+custom_hyperparameter = {    "s3_bucket": s3_bucket,
+    "s3_prefix": s3_prefix,
+    "aws_region": aws_region,
+    "model_metadata_s3_key": "%s/model/model_metadata.json" % s3_prefix,
+    "reward_function_s3_source": "%s/customer_reward_function.py" % s3_prefix,
+    "batch_size": "64",
+    "lr": "0.0003",
+    "exploration_type": "Additive_noise",
+    "e_greedy_value": "0.05",
+    "epsilon_steps": "10000",
+    "discount_factor": "0.999",
+    "sac_alpha": "0.2",
+    "stack_size": "1",
+    "loss_type": "Mean squared error",
+    "num_episodes_between_training": "20",
+    "term_cond_avg_score": "100000.0",
+    "term_cond_max_episodes": "100000"
+}
```
Should be a code block

heat-jr replied:
Done

davidfsmith commented on README.md:
```diff
@@ -1,10 +1,326 @@
+If you want to change the training algorithm to SAC, first modify the model_metadata.json file with "training_algorithm" : "sac" and a continuous action space, such as:
+
+
+{  "action_space" : {
+    "steering_angle" : {
+      "high" : 30.0,
+      "low" : -30.0
+    },
+    "speed" : {
+      "high" : 1.0,
+      "low" : 0.5
+    }
+  },
+  "sensor" : [ "FRONT_FACING_CAMERA" ],
+  "neural_network" : "DEEP_CONVOLUTIONAL_NETWORK_SHALLOW",
+  "version" : "4",
+  "training_algorithm" : "sac",
+  "action_space_type" : "continuous",
+  "preprocess_type" : null,
+  "regional_parameters" : null
+}
```
Should be a code block

heat-jr replied:
Done.

davidfsmith commented on README.md:
```diff
@@ -1,10 +1,326 @@
+If one wants to use another programming language for their reward function, one can the python boto3 library to invoke an Amazon Lambda function. Such a method may look like the following:
+
+
+import boto3,jsonlambdaservice = boto3.client('lambda')
+def reward_function(params):
+    response = lambdaservice.invoke(FunctionName='YourFunctionHere',
+                      Payload=json.dumps(params))
+        return(float(response["Payload"]))
```
Should be a code block

heat-jr replied:
Done.

heat-jr committed:
Update README.md 

Updated to add code formatting, pngs, and other formatting updates in accordance with AWS Style.

heat-jr committed:
Update README.md

Removed bug from reward function example template.

heat-jr committed:
Merge branch 'master' into heat-jr-patch-3

davidfsmith reviewed (approved):
Looks good - approving
